quil-plotting/quil_plotting/dataframe.py

```diff
@@ -0,0 +1,346 @@
+"""dataframe module.
+-----------------
+
+This module produces a plottable dataframe with IQ values and metadata.
+
+Todo:
+- Handle runner ordering
+- Use quil waveform templates
+"""
+
+import json
+from enum import Enum
+from itertools import cycle
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from pyquil.quiltwaveforms import DragGaussianWaveform, ErfSquareWaveform, FlatWaveform, GaussianWaveform
+from quil.expression import Expression, PrefixExpression, PrefixOperator
+from quil.instructions import Gate, Pragma, Pulse, Qubit, ShiftPhase
+from quil.program import Program
+
+
+# color
+class Color(str, Enum):
+    GREEN = "#00b5ad"  # Teal
+    YELLOW = "#ffc504"  # Yellow
+    RED = "#ef476f"  # Magneta
+    BLUE = "#3d47d9"  # Palatinate Blue
+    GRAY = "#8a8b92"  # Gray
+    NAVY = "#0d0d36"  # Cetacean Blue
+
+
+def hex_to_rgba(hex_color: str, opacity: float = 1.0) -> str:
+    """Convert a hex color to an rgba string.
+
+    >>> hex_to_rgba("#00B5AD")
+    'rgba(0,181,173,1.0)'
+
+    :param hex_color: The hex color code.
+    :param opacity: The opacity for the rgba color.
+    """
+    hex_color = hex_color.strip("#")
+    r, g, b = tuple(int(hex_color[i : i + 2], 16) for i in (0, 2, 4))
+    return f"rgba({r},{g},{b},{opacity})"
+
+
+def rgba_to_hex(rgba: Union[np.ndarray, List[int]]) -> str:
```

Comment:
It looks like all of these functions are public. Is that necessary? Can some of them be made private such that the public API is easier to grok and maintain?